### PR TITLE
Address uninitialized value warnings in Ubuntu 18.04

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

File: Modules/Core/Common/include/itkImageBase.hxx [hunk 1]
```diff
@@ -146,9 +144,7 @@ void
 ImageBase< VImageDimension >
 ::SetOrigin(const float origin[VImageDimension])
 {
-  Point< float, VImageDimension > of(origin);
-  PointType                       p;
-  p.CastFrom(of);
+  PointType p(origin);
```
Comment (dzenanz) [inline Modules/Core/Common/include/itkImageBase.hxx]: Isn't this the same pattern as old line 138? Does float/double distinction matter? What happens if `ITK_USE_FLOAT_SPACE_PRECISION` is enabled?

Comment (blowekamp) [reply Modules/Core/Common/include/itkImageBase.hxx]: Good catch, that can be update too.

With `ITK_USE_FLOAT_SPACE_PRECISION` enabled I get about 50% through the compilation, the some compilation errors in GaussianDerivativeImageFunction which appear to be un-releated. I don't know if this option was compilable before.

**Comment by blowekamp** [comment]
Super annoying!

File: Modules/Core/Common/include/itkImageBase.hxx (hunk 1 above)
Comment (blowekamp) [reply Modules/Core/Common/include/itkImageBase.hxx]: There are ALOT of errors related to the usage of the continuous index and interpolators. This configuration option is broken.

**Pull request merged**